scout/parse/variant/clnsig.py

```diff
@@ -58,16 +63,20 @@ def parse_clnsig(
             sig = transcripts[0].get("clinvar_clnsig")
             revstat = transcripts[0].get("clinvar_revstat")
 
-    # There are some versions where clinvar uses integers to represent terms
+    # There are some versions where ClinVar uses integers to represent terms
     if isinstance(acc, int) or acc.isdigit():
-        revstat_groups = []
-        if revstat:
-            revstat_groups = [rev.lstrip("_") for rev in revstat.replace("&", ",").split(",")]
 
-        sig_groups = []
-        for significance in sig.replace("&", ",").split(","):
-            for term in significance.lstrip("_").split("/"):
-                sig_groups.append("_".join(term.split(" ")))
+        def parse_groups(value):
+            return (
+                [item.lstrip("_") for item in value.replace("&", ",").split(",")] if value else []
+            )
+
+        revstat_groups = parse_groups(revstat)
+        sig_groups = [
+            "_".join(term.lstrip("_").split(" "))
+            for sig_term in sig.replace("&", ",").split(",")
+            for term in sig_term.split("/")
```

Comment:
Looks equivalent. To me, this is a tad harder to read, for the convoluted nesting feeling, with the incoming var appearing in the middle of the line, but all good if you find it easier - it is clearly valid notation.